frontend/src/types/auth.ts

```diff
@@ -16,3 +20,29 @@ export type ActivateInfo = {
   name: string;
   token: string;
 };
+
+export type BytebaseLoginInfo = {
+  email: string;
+  password: string;
+};
+
+export type AuthProvider = {
+  type: AuthProviderType;
+  instanceUrl: string;
+  applicationId: string;
+  secret: string;
+};
+
+export const EmptyAuthProvider: AuthProvider = {
+  type: "",
+  instanceUrl: "",
+  applicationId: "",
+  secret: "",
+};
+
+export type GitlabLoginInfo = {
```

Comment:
This can be generalized to VCSLoginInfo, likely other VCS provider will have this setup as well (also we have already added the tryLogin in the vcs plugin interface)